### ISSUE TITLE
--ignore-interfaces should not ignore builtin modules nor libraries depended on

### ISSUE THREAD

**Issue by nad** [agda/agda#2754]
The following code (an empty file) triggers an internal error if `-vtc:10 --ignore-interfaces` is used:
```agda
```
Output:
```
[...]
An internal error has occurred. Please report this as a bug.
Location of the error: src/full/Agda/TypeChecking/Level.hs:77
```

**Comment by andreasabel** [comment]
Yeah, it tries to print debug info involving levels and has not bound the built-ins yet.  In particular, it happens why it is checking the postulate for `lzero`.

I think `--ignore-interfaces` should be given an overhaul.  In particular, it should not recompile:
- the `Agda.Primitive` modules
- any library modules that come from the `depend` group rather than the `include` group

Imho, it does not make sense that if you want to recompile your project, you would also want to recompile all the libraries you depend on.

**Comment by nad** [comment]
In case of an overhaul, see also #2610 and #2330.

**Comment by nad** [comment]
Is this a duplicate of #2530?

**Comment by UlfNorell** [comment]
It looks like it is. Note also that since #3522 `--ignore-interfaces` already does not ignore builtin modules.
